src/CMakeLists.txt

```diff
@@ -54,7 +54,7 @@ else(no_yasm)
     set(CMAKE_REQUIRED_QUIET ${save_quiet})
     if(not_arch_x32)
       message(STATUS " we are not x32")
-      set(HAVE_GOOD_YASM_ELF64 1)
+      set(HAVE_GOOD_YASM_ELF64 1 PARENT_SCOPE)
```

Comment:
this breaks the build on amd64 with a capable yasm. `HAVE_GOOD_YASM_ELF64` is not defined in this scope.